### PR TITLE
Privacy mode as default

### LINKED ISSUE #6110: Add way to bypass privacy mode on a per-site basis
Privacy mode is a great improvement for how we expose identifiable information to sites, however if we roll it out today, we will break every site that is statically hosted or not actively maintained. This violates one of the basic principles of the web.

We should design & implement a method to force-sign-in to sites that have not implemented 1102 sign-in methods. Possibly a generic "sign in" button on the MetaMask popup.

### LINKED ISSUE #6320: Allow manual enable for legacy sites.
**What problem are you trying to solve?**
There are a number of older sites that do not call `.enable()`, they just try to access `web3.currentProvider`.  Without taking MetaMask out of privacy mode, I would like a way to be able to approve the site so it gets an injected `web3.currentProvider` so I can continue using the dapp.

**Describe the solution you'd like**
Put a button in the extension UI for "Allow this site to see your MetaMask address" (wording could probably be improved).

**Additional context**
Part of the idea with dapps is that they can be deployed to things like IPFS and live forever, even after the author is gone.  Also, if you audit a dapp you may not want to spend (or have the) time to re-audit it so you choose to not update to a newer version of the dapp that supports `.enable()`.  In these cases, there is value in allowing MetaMask to be backward compatible for as long as possible.

Toggling off Privacy Mode is _not_ what I want because that disables privacy mode for _every open tab I have_.  I want to just allow a specific site access through the UI.

### LINKED ISSUE #6352: Enable privacy mode by default
Privacy mode should be "ON" by default, with the option to disable.

This requires
- [ ] UI to notify users this change was made
- [ ] addition of "force log in" button described in https://github.com/MetaMask/metamask-extension/issues/6320, https://github.com/MetaMask/metamask-extension/issues/6110
- [ ] some affordance to help users on legacy sites know that they must force log-in 
  - [ ] research if we can detect sites trying to read .accounts[0] before requesting access
  - [ ] if yes, we can show a conditional UI helper just when necessary
  - [ ] if no, we should show something persistent for a period of time
- [ ] message from Twitter and Medium

### PR DESCRIPTION
**Pull request by whymarrh**
Closes #6352, fixes #6320 and fixes #6110

This PR sets Privacy Mode to be "on"/enabled by default, leaving the toggle in settings to allow folks to disable it wholly and return the to old behaviour.

**Components**

- [x] A notification is shown to users who previously had Privacy Mode disabled
- [x] For users who are already using Privacy Mode, no notice is shown
- [x] A "Share Address" notification is shown when the popup is opened on a tab that hasn't yet been connected (#6320, #6110)

**What does the migration notice look like**

Users for who Privacy Mode was previous disabled will see:

**What does the notification look like**

The design: [screenshot](https://user-images.githubusercontent.com/597770/59070279-67794f80-8877-11e9-8471-a0a319c12803.png), and [Figma](https://www.figma.com/file/foWeMNl8hQNDnxBjf2fFcbhd/privacy-mode-notice?node-id=82%3A590)

As implemented:

### CODE REVIEW

File: app/scripts/ui.js
Comment (danjm) [inline app/scripts/ui.js]: Everything below here is just moved from `popup-core.js` correct?

File: ui/app/components/app/home-notification/home-notification.container.js
Comment (danjm) [inline ui/app/components/app/home-notification/home-notification.container.js]: Is this empty mapDispatchToProps necessary?

**Review by danjm — CHANGES_REQUESTED** [review]
Only one minor requested change right now. I need to more carefully review the changes to `app/scripts/ui.js` and `app/scripts/controllers/provider-approval.js`. It seems to make sense, but I want to do another pass before final approval.

File: app/scripts/ui.js
Comment (Gudahtt) [reply app/scripts/ui.js]: In the separate notification bundle PR, I did essentially the opposite of this: I moved more things from `ui.js` into `popup-core.js`, because they were shared between `ui.js` and the newly created `notification.js`.

I'm guessing they were merged here because the separation wasn't useful? It will shortly become useful, so it might be worth moving everything back to `popup-core.js`. Oryou can not bother, and I'll just do that when resolving this conflict in my PR - I don't mind.

File: ui/app/pages/home/home.component.js
Comment (danjm) [inline ui/app/pages/home/home.component.js]: So if I read this right, this notification is only meant to be shown once, correct? And that is after the migration?

**Review by danjm — COMMENTED** [review]
Okay, everything looks good to me. I've left a couple questions and we should make a decision on the comment from Mark. If this passes QA, I will approve.

**Comment by danjm** [comment]
@bdresser Tagging you as a reviewer as you probably have the sharpest sense of how this should behave, so I think a QA from you would be good.

Also, @tmashuang, if you could test that it behaves as expected in the case of a user who has privacy mode off before this update is made, that would be good.

File: ui/app/components/app/home-notification/home-notification.container.js
Comment (whymarrh) [reply ui/app/components/app/home-notification/home-notification.container.js]: It is not, good catch!

File: app/scripts/ui.js
Comment (whymarrh) [reply app/scripts/ui.js]: > Everything below here is just moved from `popup-core.js` correct?

Yes, that file was only used here so I merged it/inlined it. The name wasn't quite accurate and it only had this single use.

I think this diff and your diff re-introducing separate files might be good together (that is, I'll leave this for now). It seems, er, semantically correct to me that we'd split things out later if needed.

File: ui/app/pages/home/home.component.js
Comment (whymarrh) [reply ui/app/pages/home/home.component.js]: Yes, once and only once

**Comment by cjeria** [comment]
The notification component styles should be consistent with the notification references in this issue https://github.com/MetaMask/metamask-extension/pull/6874 

I'm noticing that there's an opaque bg-color applied to the button that is giving it a darker color than it's parent container. 

Check out this pen to see how the styles are intended to be implemented https://codepen.io/cjeria/pen/mNmEYj

**Comment by whymarrh** [comment]
> The notification component styles should be consistent with the notification references in this issue #6874

Yes, @danjm and I are aware that the two are slightly different and we'll need to merge both their implementation and design when both PRs are landed on `develop`.

> I'm noticing that there's an opaque bg-color applied to the button that is giving it a darker color than it's parent container. Check out this pen to see [the styles]

Subtle catch, thanks. Fixed.

**Review by danjm — APPROVED** [review]
LGTM

**Pull request merged**
